File: 2do_DAM/PMDM/1ra_Evaluacion/Ejercicios/Java/Ejercicio02_ComponentesBasicos/app/src/main/java/com/example/ejercicio02_componentesbasicos/MainActivity.java
package com.example.ejercicio02_componentesbasicos;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.LinearLayout;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

import com.example.ejercicio02_componentesbasicos.databinding.ActivityMainBinding;
import com.google.android.material.snackbar.Snackbar;

import java.util.ArrayList;


public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding bilding;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //setContentView(R.layout.activity_main);

        // Esto sirve para poner el como contenido de la vista, el bilding, en la vista
        bilding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(bilding.getRoot());

        // Llamo al metodo spinnerCiudades para que cargue las ciudades
        spinnerCiudades();

        bilding.bGuardar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                recuperarDatos();
                Snackbar.make(v,"Mira el Logcat para comprobar los datos ;)", Snackbar.LENGTH_LONG).show();
            }
        });
    }
 // Creo este metodo creo un spinner, y un adaptador
    private void spinnerCiudades() {
        Spinner spinner = bilding.spinner;
        ArrayAdapter adapter = new ArrayAdapter<String>(getApplicationContext(), android.R.layout.simple_spinner_dropdown_item);
        adapter.add("Salamanca");
        adapter.add("Valladolid");
        adapter.add("Burgos");
        spinner.setAdapter(adapter);
    }

    // Recuperación de todos los datos de los objs
    private void recuperarDatos() {
        String nombre = bilding.ptNombre.getText().toString();
        String estado = ((RadioButton)findViewById(bilding.rgCS.getCheckedRadioButtonId())).getText().toString();
        String ciudad = bilding.spinner.getSelectedItem().toString();
        ArrayList<String> deportes = new ArrayList<>();

        for (int i = 0; i < bilding.checkboxGroup.getChildCount(); i++) {
            CheckBox cajaDepotes = (CheckBox) bilding.checkboxGroup.getChildAt(i);
            if (cajaDepotes.isChecked()){
                deportes.add(cajaDepotes.getText().toString());
            }
        }

        Usuario usuario = new Usuario(nombre, estado, ciudad, deportes);
        Log.d("Depurando", usuario.toString());
    }

}

File: 2do_DAM/PMDM/1ra_Evaluacion/Ejemplos/Kotlin/Ejemplo023_SharedPreferences/app/src/main/java/com/example/ejemplo023_sharedpreferences/MainActivity.kt
package com.example.ejemplo023_sharedpreferences

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.ejemplo023_sharedpreferences.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // setContentView(R.layout.activity_main)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.bGuardar.setOnClickListener {
            val nombre = binding.tietDato.text.toString()

            // Se almacenaran los datos en el fichero datos.xml
            val editor = getSharedPreferences("datos", MODE_PRIVATE)
            var preferencia = editor.edit()


            preferencia.putString("nombre", nombre)
            preferencia.apply()
            preferencia.commit()
        }

        binding.bRecuperar.setOnClickListener {

            val cadena = getSharedPreferences("dato", MODE_PRIVATE).getString("nombre", "No hay datos")
            binding.tvDatoaMostrar.text = cadena
        }
    }
}